### DISCUSSION TITLE
environment variable file (.env)

### DISCUSSION THREAD

**Discussion by fahadshery** [danielmiessler/Fabric#834] [Q&A]
Hi,

I want to use my local ollama server running on a different IP address within my IT infrastructure. 
is there an `.env` I could use to setup the IP addresses and the default model to use when installing it? 

thanks

**Comment by xssdoctor** [comment]
Yes, setting an ollama url is part of the setup. Set the url to whatever you want

**Reply by fahadshery** [reply]
> Yes, setting an ollama url is part of the setup. Set the url to whatever you want

thanks for your response. Do you have an example `.env` file? I am currently dockerising fabric so would like to it setup during build

**Comment by xssdoctor** [comment]
DEFAULT_MODEL=
PATTERNS_LOADER_GIT_REPO_URL=
PATTERNS_LOADER_GIT_REPO_PATTERNS_FOLDER=
OLLAMA_API_URL=
GEMINI_API_KEY=
GROQ_API_KEY=
ANTHROPIC_API_KEY=
OPENAI_API_KEY=
YOUTUBE_API_KEY=

Again, this is created if you run --setup

**Reply by fahadshery** [reply]
that's what I was looking for. 
I want to set it in the `.env` file and mount inside the docker container. I didn't want to run `--setup` because I would have to install without docker.
thanks for your help. I have asked the maintainer to include an empty `.env` in the project folder

**Reply by JavierHernandezG** [reply]
Hi there! did you managed to get this working? I've been trying to get fabric up using only .env since I want to skip the setup but haven't cracked it yet.

**Reply by th3f001** [reply]
I think it needs also DEFAULT_VENDOR and DEFAULT_MODEL on top of PATTERNS_LOADER_GIT_REPO_URL and PATTERNS_LOADER_GIT_REPO_FOLDER populated. For the chosen vendor the related [VENDOR]_API_KEY and [VENDOR]_API_BASE_URL must of course be populated for fabric to run.
In predefined .env scenario you might to include also FABRIC_OUTPUT_PATH to point it to a specific folder (i.e. Obisdian vault) for further integration.
